### PR TITLE
Deprecation policy for apache-airflow-providers-google package

### PR DESCRIPTION
**Pull request by VladaZakharova**
This PR announces deprecation policy for Google provider.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: docs/apache-airflow-providers-google/deprecation-policy.rst [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+          use_instead="airflow.providers.google.cloud.hooks.vertex_ai.auto_ml.AutoMLHook, "
+          category=AirflowProviderDeprecationWarning,
+      )
+
+
+  - Once the date of the deprecated method/parameter/operator is passed, it can be removed together with bumping major version of the package.
+
+Additional Considerations
+`````````````````````````
+  - Bumping a major version of the Airflow google provider package will happen once a month and everything that reached its deprecation removal date will be removed.
```
Comment (eladkal) [inline docs/apache-airflow-providers-google/deprecation-policy.rst]: We normally don't do that.
We remove code and cut breaking change release only when we have to. We don't normally cut a breaking change release  just to remove things. Release needs to have value. The contract should be users should not relay on the code after deprecation date passed regardless if we removed it or not. We remove at the first chance we have after the specified date.

**Review by potiuk — APPROVED** [review]

**Comment by moiseenkov** [comment]
Hi @potiuk , could we merge it please?

**Comment by potiuk** [comment]
> Hi @potiuk , could we merge it please?

There was a comment from @eladkal that needs to be addressed

**Comment by potiuk** [comment]
(unless it's already addressed :)

**Review by potiuk — APPROVED** [review]

**Pull request merged**